CMakeLists.txt

```diff
@@ -157,6 +165,10 @@ find_package(Qt5 COMPONENTS PrintSupport QUIET)
 find_package(QGIS REQUIRED)
 if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "iOS")
   find_package(Proj)
+  if(ANDROID)
+    # If the build target is Android, we use OSGeo4A libraries instead
+    set(ENV{GDAL_ROOT} ${OSGEO4A_STAGE_DIR}/${ANDROID_ABI})
+  endif(ANDROID)
```

Comment:
Should this go into FindGDAL instead?

Reply:
Do we have to re-implement FindGDAL from the default module or what else?

Reply:
No, I realize now, that this is actually an input for FindGDAL, sorry I missed that before.

I am wondering if all this osgeo4a specific handling is the way to go or if we should rather patch `CMAKE_FIND_ROOT_PATH` as it's being done in scripts/docker-build.sh.

A `if(${OSGEO4A_STAGE_DIR}` -> `list(APPEND CMAKE_FIND_ROOT_PATH ${OSGEO4A_STAGE_DIR}/${ANDROID_ABI}` would be a lot less intrusive over all?

Reply:
Without tinkering with `CMAKE_FIND_ROOT_PATH` CMake won't be able to locate libs in the stage path. Also, I find this snippet working instead:
```
if(OSGEO4A_STAGE_DIR)
  list(APPEND CMAKE_FIND_ROOT_PATH /)
endif()
```
If you agree I will push it.

Reply:
That looks cleaner.
Can we possibly have one single if for osgeo4a?

Reply:
I've just collected all the necessary things into `if(ANDROID)` block. Please check the latest commit.